frontend/include/chpl/libraries/LibraryFileFormat.h

```diff
@@ -57,51 +65,62 @@ struct FileHeader {
   uint32_t chplVersionUpdate;
   uint32_t nModules;
   uint8_t hash[HASH_SIZE];
-  // followed by nModules 8 bytes file offsets
+  // followed by nModules 8 bytes file offsets plus one more for
+  // just after the last module
+};
+
+struct Region {
+  uint64_t start; // start offset, maybe relative
+  uint64_t end; // end offset; so length = end-start
 };
 
+static inline Region makeRegion(uint64_t startOffset, uint64_t endOffset) {
```

Comment:
Any particular reason this isn't a constructor?

Reply:
I am always forgetting the C++ rules, but it's important that `Region` be POD (since we are "reading" it using mmap within the file). So the reason is, I wanted the C++ compiler to see it as a simple C type.